docs/design/ocs-to-ocs-api-server.md

```diff
@@ -0,0 +1,162 @@
+   google.protobuf.Struct data = 1;
+}
+
+```
+
+##### Response Scheme
+
+
+| Condition | gRPC Code | Description | Recovery |
+| -------- | -------- | -------- | -------- |
+| Success    | 0  OK     | Request Successful. StorageConsumer CR is updated, rook resources are created and CR status is updated successfully.    | N/A |
+| Awaiting Ceph resources    | 14 UNAVAILABLE     | Waiting for the ceph resourcs to be created and updated in the StoragConsumer CR status    | N/A |
+| Invalid ID | 3 INVALID_ARGUMENT | Request failed to invalid Consumer ID | Contact the Provider Cluster Admin to verify the StorageConsumer ID
+| Error | 13 INTERNAL | Request failed due to some error in provider cluster | Contact Provider Cluster Admin
+
+
+#### OffboardConsumer
+- Make a delete request on the StorageConsumer CR and return OK.
+
+```protobuf
+message OffBoardConsumerRequest{
+    // K8s UID (UUID) of the consumer cluster
+    string storageConsumerUUID =1;
+}
+
+message OffBoardConsumerResponse{
+
+}
+
+```
+
+##### Response Scheme
+| Condition | gRPC Code | Description | Recovery |
+| -------- | -------- | -------- | -------- |
+| Success    | 0  OK     | Request Successful     | N/A |
+| Invalid ID | 3 INVALID_ARGUMENT | Request failed due to invalid Consumer ID | Contact the Provider Cluster Admin to verify the StorageConsumer ID
+| Error | 13 INTERNAL | Request failed due to some error in provider cluster | Contact Provider Cluster Admin
+
+
+
+#### UpdateCapacity
+- Read the `status.state` of the StorageConsumer CR:
+  - if Status is `Ready`, then update the `spec.Capacity` and return OK
+
+```proto=
+
+message UpdateCapacityRequest{
+    // K8s UID (UUID) of the consumer cluster
+    string storageConsumerUUID =1;
+    // capacity is a valid k8s resource quantity
+    string capacity = 2;
+}
+
+message UpdateCapacityResponse{
+
+}
+```
+
+##### Response Scheme
+| Condition | gRPC Code | Description | Recovery |
+| -------- | -------- | -------- | -------- |
+| Success    | 0  OK     | Request Successful     | N/A |
+| Invalid ID | 3 INVALID_ARGUMENT | Request failed to invalid Consumer ID | Contact the Provider Cluster Admin to verify the StorageConsumer ID
+| Invalid Capacity | 3 INVALID_ARGUMENT | Request failed due to invalid capacity | Contact the Provider Cluster Admin to get the valid capacity value
+| Error | 13 INTERNAL | Request failed due to some error in the Provider Cluster | Contact Provider Cluster Admin
+
+
+
+## PROJECT STRUCTURE
+- The API server would be part of the OCS Operator.
+- It would be deployed as separate deployment on the OCS cluster only after user has enabled it in the StorageCluster CR via `spec.allowRemoteStorageConsumers`
+
+```
+.
+├── services
+    ├── provider
+        ├── pb
+            ├── provider.proto
+            ├── provider.pb.go
+            ├── README.md
+        ├── client
+            ├── client.go
+        ├── server
+            ├── server.go
+
+```
+
+## API SECURITY
```

Comment:
Empty section?

Reply:
Planning to add that in the next PR.  More discussions on this are happening [here](https://docs.google.com/document/d/1WVFslcj1SG8mAswFZLNJkq6ZLJazl9twKf2e8_79eJY/edit)

Reply:
Add a `TODO:` comment or equivalent.

Reply:
done.